api/src/test/java/org/pmiops/workbench/api/OAuth2ApiControllerTest.java

```diff
@@ -0,0 +1,114 @@
+package org.pmiops.workbench.api;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.api.client.http.HttpMethods;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.pmiops.workbench.google.CloudStorageClient;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.springframework.web.client.RestTemplate;
+
+@SpringJUnitConfig
+class OAuth2ApiControllerTest {
+  @SpyBean private OAuth2ApiController oAuth2ApiController;
+
+  @MockBean CloudStorageClient cloudStorageClient;
+  @MockBean private HttpServletRequest mockHttpServletRequest;
+  private RestTemplate restTemplate;
+
+  @TestConfiguration
+  @Import({OAuth2ApiController.class})
+  public static class Config {}
+
+  @BeforeEach
+  public void setup() throws IOException {
+    restTemplate = mock(RestTemplate.class);
+    when(oAuth2ApiController.getRestTemplate()).thenReturn(restTemplate);
+    when(mockHttpServletRequest.getMethod()).thenReturn(HttpMethods.POST);
+    when(mockHttpServletRequest.getReader()).thenReturn(new BufferedReader(new StringReader("")));
+  }
+
+  @Test
+  void testProxyOauthToken_proxiesQueryString() {
+    when(mockHttpServletRequest.getQueryString()).thenReturn("abc=123");
+    String expectedUri = String.format("%s?abc=123", OAuth2ApiController.GOOGLE_TOKEN_ENDPOINT);
+
+    oAuth2ApiController.proxyOauthToken(mockHttpServletRequest);
+
+    ArgumentCaptor<String> argument = ArgumentCaptor.forClass(String.class);
+    verify(restTemplate).exchange(argument.capture(), any(), any(), eq(JsonNode.class));
+    assertThat(argument.getValue()).isEqualTo(expectedUri);
+  }
+
+  @Test
+  void testProxyOauthToken_proxiesQueryString_empty() {
+    when(mockHttpServletRequest.getQueryString()).thenReturn("");
```

Comment:
This returns `null` if no query string, so let's test for that too

Reply:
Good catch. I added a test.